### PR TITLE
integration specs 

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ]  In this project, my partner and I created integration tests for all of the views used in my project. Be sure to cover the user stories (or user workflows) that we wanted our users to experience while taking into consideration the possible errors our users may cause.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +138,12 @@ To run tests, run the following command:
 - GitHub: [@ibtisam34](https://github.com/ibtisam34)
 - LinkedIn: [Falis](https://linkedin.com/in/falis-abdikani/)
 
+👤 **Asha S Vijayan**
```
Comment (DammyShittu) [inline README.md:141]: - [ ] Your command for running the tests is incorrect. It should be something like `rspec spec <file_name>`.

  image

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -7,15 +7,15 @@ def index
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
-    @comments = @post.comments
+    @comments = @post.comments.includes(:author)
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb]: - [ ] The purpose of eliminating the N+1 query problem, in this case, is to reduce the number of queries needed to fetch, for example, multiple resources associated with another resource. Therefore, using `.includes(:author)` here is somewhat unnecessary as there is only one author. We should use this `.includes.(:comments)` instead.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Ibtisam34 and @asharanjith,

## Highlights
✔️ Some tests pass

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Most of your tests are failing. Kindly fix them. 21 out of 30 are failing.

  image

  #### HINT: 
  You need to update your user and post creation lines. In all the spec files where you create user and posts you need to add `posts_counter` to user and `likes_counter` and `comments_counter`  to the post creation.

  ```
  User.create(name: 'pius', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', posts_counter: 0, bio: 'Dummy text for testing')

  Post.create(title: 'Hello', body: 'This is a post text', likes_counter: 0, comments_counter: 0, author_id: @user.id)
  ```
  And now all the 30 tests will pass

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @Ibtisam34,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] Some linters checks are not passed. Please make sure that all checks are passing and green.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: README.md [hunk 3]
```diff
@@ -124,7 +124,7 @@ To run the project, execute the following command:
 
 To run tests, run the following command:
 
-- `bin/rails test test/models/article_test.rb`
+- `bin/rails rspec spec <file_name>.`
```
Comment (jasonfelice) [inline README.md]: - [ ] As mentioned by the previous reviewer, please update the command to run the tests. It is still incorrect.

**Review by jasonfelice — CHANGES_REQUESTED** [review]
Hi  @Ibtisam34, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
No Linter errors✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (Abenezer-Tilahun) [reply README.md]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Ibtisam34   
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**